include/core/CPersistUtils.h

```diff
@@ -661,14 +611,12 @@ class CORE_EXPORT CPersistUtils {
             return true;
         }
 
-        // Reuse this same string to avoid as many allocations
-        // as possible.
+        // Reuse this same string to avoid as many allocations as possible.
         //
-        // The reservation is 15 because for string implementations
-        // using the short string optimisation we don't want to
-        // cause an unnecessary allocation.
+        // Reserve 17 chars because for string implementations using the short
+        // string optimisation we don't want to cause an unnecessary allocation.
```

Comment:
This comment is outdated now.

At the time it was written some platforms used the short string optimisation and others didn't.  For those that weren't it was beneficial to reserve in case the lengths of tokens went, say, 8, 9, 10, causing two reallocations.  But for strings using the short string optimisation minimum `capacity` is 15 and tokens of lengths 8, 9, 10 in that order wouldn't cause any allocations.

Now we are in a world where all platforms use the short string optimisation.  Reserving 16 or above forces Windows/Linux strings to do a heap allocation.  But 17 is not enough to avoid a second, for example if the token is `1.2345678901234567e-89`.  So I am thinking it's actually best _not_ to reserve anything.  In the case of many short numbers like `0`, `1`, `0.5` we'll avoid all allocations in this method and in the case of really long representations we might as well do the allocation when it's required.

Reply:
Good point. See [fa3ab02](https://github.com/elastic/ml-cpp/pull/1274/commits/fa3ab02793b790aa31efb0c0cd8524a645f0db6a).